--- a/xbmc/Application.cpp	
+++ b/xbmc/Application.cpp	
@@ -781,7 +781,7 @@
   return true;
 }
 
-bool CApplication::CreateGUI()
+bool CApplication::CreateGUI(bool showXBMCSplash)
 {
   m_renderGUI = true;
 #ifdef HAS_SDL
@@ -866,7 +866,7 @@
     return false;
   }
 
-  if (g_advancedSettings.m_splashImage)
+  if (g_advancedSettings.m_splashImage && showXBMCSplash)
   {
     CStdString strUserSplash = "special://home/media/Splash.png";
     if (CFile::Exists(strUserSplash))

--- a/xbmc/Application.h	
+++ b/xbmc/Application.h
@@ -129,7 +129,7 @@
   virtual bool Create();
   virtual bool Cleanup();
 
-  bool CreateGUI();
+  bool CreateGUI(bool showXBMCSplash);
   bool InitWindow();
   bool DestroyWindow();
   void StartServices();

--- a/xbmc/xbmc.cpp	
+++ b/xbmc/xbmc.cpp
@@ -20,6 +20,8 @@
 
 #include "Application.h"
 #include "settings/AdvancedSettings.h"
+#include "stdlib.h"
+#include <sys/stat.h>
 
 #ifdef TARGET_RASPBERRY_PI
 #include "linux/RBP.h"
@@ -27,7 +29,15 @@
 
 extern "C" int XBMC_Run(bool renderGUI)
 {
-  int status = -1;
+  int status = -1;    
+  bool showXBMCSplash = true;
+  std::string filename = "/usr/bin/splash";
+  struct stat buf;
+  if (stat(filename.c_str(), &buf) != -1)
+  {
+    system("sudo /usr/bin/splash --infinitebar --msgtxt='starting xbmc...'");
+    showXBMCSplash = false;
+  }
 
   if (!g_advancedSettings.Initialized())
   {
@@ -53,7 +63,7 @@
   g_RBP.LogFirmwareVerison();
 #endif
 
-  if (renderGUI && !g_application.CreateGUI())
+  if (renderGUI && !g_application.CreateGUI(showXBMCSplash))
   {
     fprintf(stderr, "ERROR: Unable to create GUI. Exiting\n");
     return status;
@@ -63,6 +73,10 @@
     fprintf(stderr, "ERROR: Unable to Initialize. Exiting\n");
     return status;
   }
+    
+  if (!showXBMCSplash) {
+    system("sudo /usr/bin/splash --black");
+  }
 
   try
   {

--- a/xbmc/ApplicationMessenger.cpp	
+++ b/xbmc/ApplicationMessenger.cpp	
@@ -21,6 +21,7 @@
 #include "system.h"
 #include "ApplicationMessenger.h"
 #include "Application.h"
+#include "stdlib.h"
 
 #include "PlayListPlayer.h"
 #include "Util.h"
@@ -1073,6 +1074,7 @@
 
 void CApplicationMessenger::Quit()
 {
+  system("sudo /usr/bin/splash --exit");
   ThreadMessage tMsg = {TMSG_QUIT};
   SendMessage(tMsg);
 }

