--- xbmc/windows/GUIWindowSystemInfo.cpp	2012-12-01 21:28:32.000000000 +0100
+++ xbmc/windows/GUIWindowSystemInfo.cpp	2012-12-01 21:36:49.000000000 +0100
@@ -18,6 +18,8 @@
  *
  */
 
+#include <iostream>
+#include <fstream>
 #include "system.h"
 #include "GUIWindowSystemInfo.h"
 #include "GUIInfoManager.h"
@@ -85,6 +87,22 @@ bool CGUIWindowSystemInfo::OnMessage(CGU
   return CGUIWindow::OnMessage(message);
 }
 
+std::string getXBianVer()
+{
+    std::ifstream verFile;
+    std::string ver;
+    verFile.open("/etc/xbian_version");
+    
+    if (verFile.is_open()) {
+        while (!verFile.eof()) {            
+            verFile >> ver;            
+        }
+    }
+    
+    verFile.close();
+    return ver;
+}
+
 void CGUIWindowSystemInfo::FrameMove()
 {
   ResetLabels();
@@ -100,7 +118,7 @@ void CGUIWindowSystemInfo::FrameMove()
 #endif
     SetControlLabel(i++, "%s: %s", 12390, SYSTEM_UPTIME);
     SetControlLabel(i++, "%s: %s", 12394, SYSTEM_TOTALUPTIME);
-    SetControlLabel(i++, "%s: %s", 12395, SYSTEM_BATTERY_LEVEL);
+    SetControlLabel(i++, "XBian version: %s", getXBianVer());
   }
   else if (m_section == CONTROL_BT_STORAGE)
   {
