--- xbmc/utils/SystemInfo.cpp	2012-12-01 17:50:13.000000000 +0100
+++ xbmc/utils/SystemInfo.cpp	2012-12-01 19:58:06.000000000 +0100
@@ -61,7 +61,7 @@ bool CSysInfoJob::DoWork()
   m_info.cpuFrequency      = GetCPUFreqInfo();
   m_info.kernelVersion     = CSysInfo::GetKernelVersion();
   m_info.macAddress        = GetMACAddress();
-  m_info.batteryLevel      = GetBatteryLevel();
+  m_info.batteryLevel      = GetXBianVer();
   return true;
 }
 
@@ -104,11 +104,19 @@ CStdString CSysInfoJob::GetVideoEncoder(
   return "GPU: " + g_Windowing.GetRenderRenderer();
 }
 
-CStdString CSysInfoJob::GetBatteryLevel()
+CStdString CSysInfoJob::GetXBianVer()
 {
-  CStdString strVal;
-  strVal.Format("%d%%", g_powerManager.BatteryLevel());
-  return strVal;
+  CStdString verVal;
+  std::ifstream verFile;
+  verFile.open("/Users/Koenkk/Desktop/xbian_version");
+  if (verFile.is_open()) {
+      while (!verFile.eof()) {
+          verFile >> verVal;
+      }
+  }
+  verFile.close();
+    
+  return verVal;
 }
 
 double CSysInfoJob::GetCPUFrequency()

--- xbmc/utils/SystemInfo.h	2012-12-01 17:50:13.000000000 +0100
+++ xbmc/utils/SystemInfo.h	2012-12-01 19:54:43.000000000 +0100
@@ -71,7 +71,7 @@ private:
   CStdString GetCPUFreqInfo();
   CStdString GetMACAddress();
   CStdString GetVideoEncoder();
-  CStdString GetBatteryLevel();
+  CStdString GetXBianVer();
 
   CSysData m_info;
 };

--- xbmc/windows/GUIWindowSystemInfo.cpp	2012-12-01 17:50:14.000000000 +0100
+++ xbmc/windows/GUIWindowSystemInfo.cpp	2012-12-01 20:18:23.000000000 +0100
@@ -100,7 +100,7 @@ void CGUIWindowSystemInfo::FrameMove()
 #endif
     SetControlLabel(i++, "%s: %s", 12390, SYSTEM_UPTIME);
     SetControlLabel(i++, "%s: %s", 12394, SYSTEM_TOTALUPTIME);
-    SetControlLabel(i++, "%s: %s", 12395, SYSTEM_BATTERY_LEVEL);
+    SetControlLabel(i++, "%s: %s", "XBian version", SYSTEM_BATTERY_LEVEL);
   }
   else if (m_section == CONTROL_BT_STORAGE)
   {
