
--- configure.ac1       2012-11-15 14:31:24.448087379 +0100
+++ configure.ac        2012-11-15 14:32:02.977755311 +0100
@@ -171,8 +171,7 @@
        (lirc_dev lirc_parallel) \
        (lirc_dev lirc_sasem) \
        (lirc_dev lirc_serial) \
-       (lirc_dev lirc_sir) \
-       (lirc_dev lirc_wpc8769l)"
+       (lirc_dev lirc_sir)"
 fi

 AH_TEMPLATE([DAEMONIZE],
@@ -608,8 +607,6 @@
                ;;
        lirc_dev-lirc_sir)
                ;;
-       lirc_dev-lirc_wpc8769l)
-               ;;
        lirc_flyvideo)
                ;;
        livedrive_midi)
@@ -1402,12 +1399,6 @@
   lircd_conf="adstech/lircd.conf.usbx-707"
 fi

-if test "$driver" = "wpc8769l"; then
-  lirc_driver="lirc_dev lirc_wpc8769l"
-  lircd_conf="acer/lircd.conf.Aspire_6530G"
-fi
-
-
 #END HARDWARE HEURISTIC

 AC_DEFINE_UNQUOTED(HW_DEFAULT, ${HW_DEFAULT})
@@ -1512,9 +1503,9 @@
        lirc_sasem \
        lirc_serial \
        lirc_sir \
-       lirc_ttusbir \
-       lirc_wpc8769l"
+       lirc_ttusbir"
 fi
+
 if kernel_module=$(expr "$lirc_driver" : 'lirc_dev \(.*\)'); then
   :
 elif test "$lirc_driver" != ""; then
@@ -1779,7 +1770,6 @@
   drivers/lirc_sasem/Makefile
   drivers/lirc_serial/Makefile
   drivers/lirc_sir/Makefile
-  drivers/lirc_wpc8769l/Makefile
   daemons/Makefile
   tools/Makefile
   doc/Makefile